integration/cmd/osrm-traffic-updater/speed_table_dumper.go

```diff
@@ -94,21 +116,41 @@ func task(wayid2speed map[int64]int, source <-chan string, sink chan<- string, d
 				}
 				if okFwd {
 					fwdRecordCnt += 1
-					sink <- generateSingleRecord(n1, n2, speedFwd, true)
+					sink <- generateSingleRecord(n1, n2, speedsFwd[i], true)
 				}
 				if okBwd {
 					bwdRecordCnt += 1
-					sink <- generateSingleRecord(n1, n2, speedBwd, false)
+					sink <- generateSingleRecord(n1, n2, speedsBwd[i], false)
 				}
 
 			}
 		}
 	}
 
-	ds.Update(wayCnt, nodeCnt, fwdRecordCnt, bwdRecordCnt, wayMatched, nodeMatched, fwdTrafficMatched, bwdTrafficMatched)
+	ds.Update(wayCnt, nodeCnt, fwdRecordCnt, bwdRecordCnt, wayMatched, nodeMatched, fwdTrafficMatched, bwdTrafficMatched, skippedSegmentsCnt)
 	tasksWg.Done()
 }
 
+func getSpeedOfSegments(segments []*trafficproxy.SegmentedFlow, speeds []int, nodesCnt uint64, skippedSegmentsCnt *uint64) {
+	for _, segment := range segments {
+		if 0 > segment.Begin || segment.Begin > segment.End || segment.End > 100 {
+			glog.Warningf("Unexpected segment length begin: %v, end: %v, should be between 0..100 \n", segment.Begin, segment.End)
+			continue
+		}
+
+		indexOfBegin := int(math.Floor(float64(nodesCnt) * float64(segment.Begin) / 100))
+		indexOfEnd := int(math.Floor(float64(nodesCnt) * float64(segment.End) / 100))
```

Comment:
Here might be some problem due to floating calculation. E.g., totally `11` nodes and `3` segmented flows `{Speed: 10, Begin: 0, End: 27},{Speed: 25, Begin: 28, End: 75},{Speed: 60, Begin: 76, End: 100}` on a way, and the segmented flows will be applied to node `0~2`, `3~8`, `8~11`, but the `2~3` is missed.      
To solve the problem, it might be better to sort the segmented flows of the way first, then adjust the nodes offset, i.e. `3~8` -> `2~8`.

Reply:
@wangyoucao577 I got it. But maybe it is ok.

For example, if first `End` and  second `Begin` have the same osm nodeID, the segments will be  `{Speed: 10, Begin: 0, End: 27}, {Speed: 20, Begin: 27, End: 75}`. It works for the same dataset on each side (traffic-updater and traffic-proxy).  

On the other hand, I could add this condition  
```
if lastEnd - curBegin == 1 {
	indexOfBegin + 1
}
```
But it will affect all segments  i.e `8~11` -> `7~11`

May be

```
if lastEnd - curBegin == 1 && lastIndexOfEnd != indexOfBegin {
	indexOfBegin + 1
}
```

Reply:
Oh! Maybe this issue will gone if we simply to make sure `next begin == last end`. I.e.,    
first segment: `{Speed: 10, Begin: 0, End: 27}`, then `{Speed: 25, Begin: 27, End: 75}` instead of `{Speed: 25, Begin: 28, End: 75}` as second segment(`Begin: 28` ->`Begin: 27`).        
How's your opinion?